src/Core/Domain/Customer/ValueObject/ApeCode.php

```diff
@@ -40,7 +40,7 @@ class ApeCode
     private $code;
 
     /**
-     * @param $code
+     * @param string $code
```

Comment:
`$code` real type is `mixed`. We want it to be a string 😄  but we are not sure which is why we assert it. We expect sometimes people pass something else than a string here.

Reply:
Wrong. The class `ApeCode` is used twice :
- src/Core/Domain/Customer/Command/AddCustomerCommand.php
- src/Core/Domain/Customer/Command/AddCustomerCommand.php

In each case, the parameter is in string. And for each use of setApeCode, data are casted in string.

Reply:
It is not enough to check that `ApeCode` _usage_ is always strings, we must also expect that `ApeCode` will be used
- in code that will be written by outside people using our code as API (module and shop developers)
- in code that will be written in the future by us (although we could consider we will be careful)

The class `ApeCode` must behave like a stand-alone class that might be used outside of our context. It's built to be used and extended by module and shop developers. It must contain the necessary checks to ensure its own integrity because we cannot rely on the discipline of our users 😅 .